pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -407,6 +406,30 @@ static char *int64ToIso(int64_t value, NPY_DATETIMEUNIT base, size_t *len) {
     return result;
 }
 
+/* Converts the int64_t representation of a duration to ISO; mutates len */
+static char *int64ToIsoDuration(int64_t value, size_t *len) {
```

Comment:
I think after this PR going to split these conversion functions off into a separate module as the JSON one is getting rather big

Reply:
The other PR is #31057 should probably be merged first